calico_containers/tests/st/workload.py

```diff
@@ -34,6 +34,10 @@ def __init__(self, host, name, ip=None, image="busybox", network=None):
         """
         self.host = host
         self.name = name
+        try:
+            host.execute('docker rm -f %s' % name)
```

Comment:
I don't think workloads should try to destroy anything existing with their name.  This is going to lead to hard-to-diagnose test script bugs.

Two ways to fix this instead:

1. Fix `DockerHost` to track and delete workloads when it gets shut down.
2. Stop manually naming workloads altogether.  There just isn't a need now that they are Python Objects.  Let Docker name the workload and `docker inspect` for the name at creation.  Then if the script really needs the name, it can use the `.name` field.

It would be good to do both, actually.